sql/catalyst/src/main/scala/org/apache/spark/sql/types/dataTypes.scala

```diff
@@ -402,7 +402,7 @@ case object DateType extends NativeType {
 }
 
 
-protected[sql] abstract class NumericType extends NativeType with PrimitiveType {
```

Comment:
Seems there are users who need it (https://issues.apache.org/jira/browse/SPARK-5288).